qcodes/instrument/parameter.py

```diff
@@ -94,84 +93,132 @@ class _BaseParameter(Metadatable, DeferredOperations):
             ``update=True``, for example if it takes too long to update.
             Default True.
 
+        snapshot_value (Optional[bool]): False prevents parameter value to be 
+            stored in the snapshot. Useful if the value is large.
+            
+        step (Optional[Union[int, float]]): max increment of parameter value.
+            Larger changes are broken into multiple steps this size.
+            When combined with delays, this acts as a ramp.
+            
+        scale (Optional[float]): Scale to multiply value with before 
+            performing set. the internally multiplied value is stored in 
+            `raw_value`. Can account for a voltage divider.
+        
+        inter_delay (Optional[Union[int, float]]): Minimum time (in seconds) 
+            between successive sets. If the previous set was less than this, 
+            it will wait until the condition is met. 
+            Can be set to 0 to go maximum speed with no errors.
+        
+        post_delay (Optional[Union[int, float]]): time (in seconds) to wait 
+            after the *start* of each set, whether part of a sweep or not. 
+            Can be set to 0 to go maximum speed with no errors.
+        
+        val_mapping (Optional[dict]): a bidirectional map data/readable values
+            to instrument codes, expressed as a dict:
+            ``{data_val: instrument_code}``
+            For example, if the instrument uses '0' to mean 1V and '1' to mean
+            10V, set val_mapping={1: '0', 10: '1'} and on the user side you
+            only see 1 and 10, never the coded '0' and '1'
+            If vals is omitted, will also construct a matching Enum validator.
```

Comment:
As far as I can see, the "matching `Enum` validator" is never set, and the parameter thus gets the default `Numbers` validator. This breaks (amongst others) a handful of parameters using a `{'ON': 1, 'OFF': 0}` val mapping.